envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/filters/RBACFilterFactory.kt

```diff
@@ -0,0 +1,102 @@
+package pl.allegro.tech.servicemesh.envoycontrol.snapshot.listeners.filters
```

Comment:
Side note: here are validations for RBAC in Envoy: https://github.com/envoyproxy/envoy/blob/5248a4fb7d4c2a3d1fa151f944d3a63f6b7a06cf/test/extensions/filters/common/rbac/engine_impl_test.cc

Here is a short list of what I got from that:
Rbac validation:
- >0 permissions
- permissions.rule is required
- andRules: rule is required
- >0 principals, has identifier (one of and_ids, or_ids, any, authenticated, source_ip, header, metadata, not_id)